packages/cadl-ranch-specs/http/projection/client-encoded-name/main.tsp

```diff
@@ -0,0 +1,178 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Projection")
+@supportedBy("dpg")
+@scenarioService("/projection/client-name-and-encoded-name")
+namespace Projection.ClientNameAndEncodedName;
+
+@route("/property")
+@operationGroup
+namespace Property {
+  model LanguageClientNameAndEncodedNameModel {
+    @doc("Pass in true")
+    @clientName("CSName", "csharp")
+    @clientName("GoName", "go")
+    @clientName("JavaName", "java")
+    @clientName("TSName", "javascript")
+    @clientName("python_name", "python")
+    defaultName: boolean;
+  }
+
+  model JsonClientNameAndEncodedNameModel {
+    @doc("Pass in true")
+    @encodedName("application/json", "wireName")
+    defaultName: boolean;
+  }
+
+  model ClientClientNameAndEncodedNameModel {
+    @doc("Pass in true")
+    @clientName("clientName")
+    defaultName: boolean;
+  }
+
+  model JsonAndClientClientNameAndEncodedNameModel {
+    @doc("Pass in true")
+    @clientName("clientName")
+    @encodedName("application/json", "wireName")
+    defaultName: boolean;
+  }
+
+  @scenario
+  @scenarioDoc("""
+  Testing that we can project the JSON name on the wire from defaultName -> wireName.
+  Your generated SDK should generate JsonClientNameAndEncodedNameModel with one property `defaultName` with wire name `wireName`.
+
+  Expected request body:
+  ```json
+  {"wireName": true}
+  ```
+  """)
+  @route("/json")
+  @post
+  op json(...JsonClientNameAndEncodedNameModel): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+Testing that we can project the client name in our generated SDKs.
+Your generated SDK should generate ClientClientNameAndEncodedNameModel with one property `clientName` with wire name `defaultName`.
+
+Expected request body:
+```json
+{"defaultName": true}
```

Comment:
Is this correct?

Maybe double check the docs?

Reply:
i think it is no? property is `defaultName` and it has a `@clientName("clientName")` so over json it is `defaultName`

Reply:
Timothee you are correct. Yeah it does not have encodedName. `JsonAndClientClientNameAndEncodedNameModel` is the one.